File: core/src/io/cucumber/model/characters/Enemy.java
package io.cucumber.model.characters;

import com.badlogic.gdx.graphics.Texture;
import com.badlogic.gdx.graphics.g2d.Batch;

import io.cucumber.model.base.DynamicActor;

public class Enemy extends DynamicActor {

    private byte orientation;

    public Enemy(float x, float y, float size, float horizontalVelocity, Texture texture, byte orientation) {
        super(x, y, size, horizontalVelocity, 0, texture);
        this.orientation = orientation;
    }

    @Override
    public void draw(Batch batch, float parentAlpha) {
        batch.draw(
                currentTexture,
                getX(),
                getY(),
                getWidth(),
                getHeight()
        );
    }

    @Override
    public void act(float delta) {
        setX(getX() + velocity.x * -1 * orientation * delta);
    }
}

File: core/src/io/cucumber/model/Enemy.kt
package io.cucumber.model

import com.badlogic.gdx.graphics.Texture
import io.cucumber.model.base.Actor

class Enemy(
    x: Float,
    y: Float,
    size: Float,
    horizontalVelocity: Float,
    texture: Texture,
    private val orientation: Byte
) : Actor(x, y, size, horizontalVelocity, 0F, texture) {

    override fun update(delta: Float) {
        position.add(velocity.x * -1 * orientation * delta, 0F)
    }

}